Add for-end-users section in Milkomeda Rollup Docs (#8)
dcSpark/milkomeda-documentation: docs/algorand/for-end-users/wrapping-assets.mdx

---
sidebar_position: 3
---
import dappselectorigin from "@site/static/img/dappselectorigin.png";
import wrappingsummary from "@site/static/img/wrappingsummary.png"; 
import wrappingcompleted from "@site/static/img/wrappingcompleted.png";
import wrapsuccess from "@site/static/img/wrapsuccess.png"; 

# Wrapping Assets
<p style={{textAlign:"justify"}}>

> In order to **transfer Assets from Algorand (mainchain) to Milkomeda A1 Rollup **, you need to use dedicated Bridge dApp.

Reviewer: ```suggestion
In order to **transfer assets from Algorand (mainchain) to the Milkomeda A1 Rollup **, you need to use our dedicated Bridge dApp.
```

> To wrap your assets onto A1 Rollup go to the Milkomeda A1 bridge dApp at https://algorand-bridge-dev.milkomeda.com/.

Reviewer: ```suggestion
To wrap your assets onto the A1 Rollup go to the Milkomeda A1 bridge dApp at [https://algorand-bridge-dev.milkomeda.com/](https://algorand-bridge-dev.milkomeda.com/).
```

Reviewer: Dev net link is used here but we're talking about the mainnet procedure

> To start wrapping your assets, choose **Algorand to Milkomeda** direction from **Network Origin** dropdown button. 

Reviewer: ```suggestion
To start wrapping your assets, choose the **Algorand to Milkomeda** direction from the "Network Origin" dropdown menu. 
```

> Then select the token you want to wrap ( in this tutorial, we will choose the ALGO token) from **Select Token** dropdown button. 

Reviewer: ```suggestion
Then select the token you want to wrap ( in this tutorial, we will use ALGO) from the "Select Token" dropdown menu. 
```

<img src={dappselectorigin} className="img-full" alt="wrapping confirmation" />

> The next step will be to log in to the MyAlgo and Metamask wallets. 

Reviewer: ```suggestion
The next step is to log in to your MyAlgo and MetaMask wallets. 
```

<ol>

> <li>Click “Connect Wallet” on the Origin side and your MyAlgo wallet (we recently also added Pera Wallet support) will require you to enter the password and confirm which account to connect to the dApp. 

Reviewer: ```suggestion
<li>Click “Connect Wallet” on the Origin side and your MyAlgo wallet (we recently added support for Pera wallet too) will require you to enter your password and confirm which account to connect to the dApp. 
```

Your available ALGO balance will then be displayed below the select token field.</li>

> <li>Click “Connect Wallet” on the Destination side of the dApp and your Metamask wallet will ask you to confirm that you want to switch networks (if you are not already connected to it) to the Milkomeda A1 network. Then you need to select and confirm which account connect to the dApp.</li>

Reviewer: ```suggestion
<li>Click “Connect Wallet” on the Destination side of the dApp and your Metamask wallet will ask you to confirm that you want to switch networks (if you are not already connected to it) to the Milkomeda A1 network. You then need to select and confirm which account connect to the dApp.</li>
```

</ol>

**Enter how many test ALGOs** you want to transfer to the A1 rollup (if you want to send your whole balance click “Max” to have the bridge calculate the maximum amount you can send minus the transaction fee). 

**Click “Next”** to view a transaction summary, complete with “Estimated Fees” and an “Estimated Wrapping Confirmation Time”. 

<img src={wrappingsummary} className="img-full" alt="wrapping summary" />

> **Click “Sign and Send”** , enter your wallet password and confirm and then watch as your tokens get sent to the bridge, wrapped, and sent to your Metamask wallet on the Milkomeda Algorand Rollup! The process should take around 20 seconds to confirm.

Reviewer: ```suggestion
**Click “Sign and Send”**, enter your wallet password, and hit confirm. Now watch as your tokens get sent to the bridge, wrapped, and sent to your MetaMask wallet on the Milkomeda | Algorand A1 Rollup! The process should take around 20 seconds to confirm.
```

:::info

> Remember that transactions from the Rollup are written to the Algorand Testnet blockchain itself. One confirmation is for sending the funds from the wallet to the bridge address and the second confirmation is for the rollup batch transaction on Algorand.

Reviewer: Say testnet but we're talking about mainnet

> Remember that transactions from the Rollup are written to the Algorand Testnet blockchain itself. One confirmation is for sending the funds from the wallet to the bridge address and the second confirmation is for the rollup batch transaction on Algorand.

Reviewer: ```suggestion
Remember that transactions from the Rollup are written to the Algorand Testnet blockchain itself. One confirmation is for sending funds from the wallet to the bridge address and the second confirmation is for the rollup batch transaction on Algorand.
```

:::

> The transfer only requires a total of 2 confirmations, which should take approximately 20 seconds (4.5 seconds for Algorand and around 14 seconds for the batch of the Rollup). Once the transaction is confirmed, the transferred MilkTALGO should be visible in your Milkomeda Algorand Testnet wallet in MetaMask.

Reviewer: ```suggestion
The transfer only requires a total of 2 confirmations, which should take approximately 20 seconds (4.5 seconds for Algorand and around 14 seconds for the posting of the Rollup batch). Once the transaction is confirmed, the transferred MilkTALGO should be visible in your Milkomeda Algorand Testnet wallet in MetaMask.
```

Reviewer: Again, this is mainnet. You say testnet

<img src={wrappingcompleted} className="img-full" alt="wrapping completed" />

You can now watch your transaction pass through the bridge.
<img src={wrapsuccess} className="img-full" alt="wrapping success popup" />

> **Congratulations!** You just wrapped your tokens onto the rollup outside of the Ethereum ecosystem!

Reviewer: ```suggestion
**Congratulations!** You just wrapped your tokens onto on of the first rollups outside of the Ethereum ecosystem!
```

> Now that we have successfully wrapped Test ALGO into MilkTALGO for use on the Milkomeda A1 Algorand Rollup Testnet, let's see how to move assets back from Milkomeda A1 Rollup to Algorand. 

Reviewer: ```suggestion
Now that we have successfully wrapped Test ALGO into MilkTALGO, for use on the Milkomeda A1 Algorand Rollup Testnet, let's see how to move assets from Milkomeda A1 Rollup to Algorand. 
```
